packages/grid/x-data-grid/src/joy/joySlots.tsx

```diff
@@ -415,7 +451,11 @@ const joySlots: UncapitalizeObjectKeys<Partial<GridSlotsComponent>> = {
   baseSelect: Select,
   baseSelectOption: Option,
   baseInputLabel: InputLabel,
-  baseFormControl: JoyFormControl,
+  baseFormControl: FormControl,
+  baseMenu: Menu,
+  baseMenuItem: JoyMenuItem,
+  baseListItemIcon: JoyListItemDecorator,
+  baseListItemText: React.Fragment,
```

Comment:
This could be `ListItemContent`
```suggestion
  import ListItemContent from '@mui/joy/ListItemContent';
  ...
  baseListItemText: ListItemContent,
```